src/libsync/discovery.cpp

```diff
@@ -405,7 +405,7 @@ bool ProcessDirectoryJob::handleExcluded(const QString &path, const Entries &ent
                 item->_errorString = tr("File names ending with a period are not supported on this file system.");
             } else {
                 char invalid = '\0';
-                foreach (char x, QByteArray("\\:?*\"<>|")) {
+                for (char x : QByteArray("\\:?*\"<>|")) {
```

Comment:
I think it would be better to define the QByteArray as a constexpr outside of the loop, maybe with a QByteArrayView